src/main/java/com/target/studentinfo/controller/StudentController.java

```diff
@@ -1,59 +1,52 @@
 package com.target.studentinfo.controller;
-
+import com.target.studentinfo.dto.mapper.StudentMapper;
 import com.target.studentinfo.dto.request.StudentRequest;
 import com.target.studentinfo.dto.response.StudentResponse;
 import com.target.studentinfo.model.Student;
 import com.target.studentinfo.service.StudentService;
 import org.springframework.web.bind.annotation.*;
-
 import java.util.List;
 import java.util.Optional;
-import java.util.UUID;
 
 @RestController
-@RequestMapping("/student_information_service/v1")
 public class StudentController {
     private final StudentService studentService;
+    private final StudentMapper studentMapper;
 
-    public StudentController(StudentService studentService) {
+    public StudentController(StudentService studentService , StudentMapper studentMapper  ) {
         this.studentService = studentService;
+        this.studentMapper = studentMapper;
     }
 
     @GetMapping("/students")
-    public List<Student> getAllStudentInfo(){
-
-        return studentService.getAllStudentInfo();
+    public List<StudentResponse> getAllStudentInfo(@RequestParam(defaultValue="true") Boolean isActive){
+        List<Student> student = studentService.getAllStudentInfo(isActive);
+        return studentMapper.StudentDetails(student);
     }
 
-    @GetMapping("/student/{id}")
-    public Optional<Student> getStudentById(@PathVariable("id") UUID id){
-        return studentService.getStudentById(id);
+    @GetMapping("/students/{id}")
+    public StudentResponse getStudentById(@PathVariable("id") long id , @RequestParam(defaultValue="true") Boolean isActive){
+        Optional<Student> student = studentService.getStudentById(id, isActive);
+        return studentMapper.StudentDetail(student);
     }
 
-    @PostMapping("/student")
-    public Student addNewStudent(@RequestBody Student student){
-        return studentService.addNewStudent(student);
+    @PostMapping("/students")
+    public StudentResponse addStudent(@RequestBody StudentRequest studentRequest){
+        Student student = studentMapper.toStudent(studentRequest);
+        studentService.addStudent(student);
+        return studentMapper.toStudentResponse(student);
     }
 
-    @PutMapping("/student/{id}")
-    public Student updateStudent(@RequestBody Student student) {
-        return studentService.updateStudent(student);
+    @PutMapping("/students/{id}")
+    public StudentResponse updateStudent(@RequestBody StudentRequest studentRequest, @PathVariable("id") long id) {
+        Student student = studentMapper.toStudent(studentRequest);
+        student.setId(id);
+        studentService.updateStudent(student);
+        return studentMapper.toStudentResponse(student);
```

Comment:
Even though you're making a service call, you're not sending it's returned value to the mapper.
Maybe you should do something like:
```
Student updatedStudent = studentService.updateStudent(student);
return studentMapper.toStudentResponse(updatedStudent);
```

Reply:
One more suggestion is, send the id also to service layer for updateStudent like
```
Student updatedStudent = studentService.updateStudent(student, id);
```
and then set the id in updateStudent() function in service layer
```
    public Student updateStudent(Student student, long id) {
        student.setId(id);
        return studentRepository.save(student);
    }
```